test/cat.jl

```diff
@@ -120,11 +120,13 @@ module TestCat
     @test size(vcat(df, df, df)) == (size(df, 1) * 3, size(df, 2))
 
     alt_df = deepcopy(df)
-    vcat(df, alt_df)
+    @test vcat(df, alt_df) == DataFrame([[3.0,2.0,3.0,3.0,3.0,2.0,3.0,3.0],
+                                         [2.0,2.0,1.0,3.0,2.0,2.0,1.0,3.0],
+                                         [2,2,2,3,2,2,2,3]])
 
     # Don't fail on non-matching types
     df[1] = zeros(Int, nrow(df))
-    vcat(df, alt_df)
+    @test size(vcat(df, alt_df)) == (8, 3)
```

Comment:
Could you also test the result here? That would also make sense below even if that's cumbersome (maybe we don't need to test all long combinations if they are redundant). Else coverage makes it look like we test some code while we actually only check that it doesn't throw an error, which is misleading.

Reply:
added